axon-server-connector/src/main/java/org/axonframework/axonserver/connector/heartbeat/HeartbeatConnectionCheck.java

```diff
@@ -0,0 +1,89 @@
+package org.axonframework.axonserver.connector.heartbeat;
+
+import org.axonframework.axonserver.connector.AxonServerConnectionManager;
+
+import java.time.Clock;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Consumer;
+
+import static io.axoniq.axonserver.grpc.control.PlatformOutboundInstruction.RequestCase.HEARTBEAT;
+
+/**
+ * Implementation of {@link ConnectionSanityCheck} which verifies that heartbeat is properly received.
+ *
+ * @author Sara Pellegrini
+ * @since 4.2
+ */
+public class HeartbeatConnectionCheck implements ConnectionSanityCheck {
+
+    private final ConnectionSanityCheck delegate;
+
+    private final long heartbeatTimeout;
+
+    private final Clock clock;
+
+    private final AtomicReference<Instant> lastReceivedHeartbeat = new AtomicReference<>();
+
+    /**
+     * Constructs an instance of {@link HeartbeatConnectionCheck} using the specified parameters.
+     *
+     * @param connectionManager the connectionManager to AxonServer instance
+     * @param context           the context
```

Comment:
`the context` does not really explain what this parameter means to be honest. I think this could be a little more descriptive.